scale-decode/examples/visitor.rs

```diff
@@ -47,143 +48,153 @@ enum Value {
 }
 
 // Implement the `Visitor` trait to define how to go from SCALE
-// values into this type:
-struct ValueVisitor;
-impl visitor::Visitor for ValueVisitor {
+// values into this type. Here, we are choosing to be generic over
+// how types are resolved, which is a good default choice when creating
+// a visitor unless you rely on a specific type resolver/ID type.
+struct ValueVisitor<R>(PhantomData<R>);
+
+impl<R> ValueVisitor<R> {
+    fn new() -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<R: TypeResolver> visitor::Visitor for ValueVisitor<R> {
     type Value<'scale, 'info> = Value;
     type Error = visitor::DecodeError;
+    type TypeResolver = R;
```

Comment:
The user facing change is basically to add this associated type, and normally to just be generic over it since you probably don't care what it is. Then, update the type signatures like below, but the logic normally stays the same.